packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,363 @@
+    return new IndySdkToAskarMigrationUpdater(
+      store,
+      walletConfig.id,
+      agent,
+      dbPath,
+      deleteOnFinish,
+      defaultLinkSecretId
+    )
+  }
+
+  /*
+   * Checks whether the destination locations are allready used. This might
+   * happen if you want to migrate a wallet when you already have a new wallet
+   * with the same id.
+   */
+  private async assertDestinationsAreFree() {
+    const areAllDestinationsTaken =
+      (await this.fs.exists(this.backupFile)) || (await this.fs.exists(this.newWalletPath))
+
+    if (areAllDestinationsTaken) {
+      throw new IndySdkToAskarMigrationError(
+        `Files already exist at paths that will be used for backing up. Please remove them manually. Backup path: '${this.backupFile}' and new wallet path: ${this.newWalletPath} `
+      )
+    }
+  }
+
+  /**
+   * Location of the new wallet
+   */
+  private get newWalletPath() {
+    return `${this.fs.dataPath}/wallet/${this.walletId}/sqlite.db`
+  }
+
+  /**
+   * Temporary backup location of the pre-migrated script
+   */
+  private get backupFile() {
+    return `${this.fs.tempPath}/${this.walletId}.bak.db`
+  }
+
+  /**
+   * Backup the database file. This function makes sure that the the indy-sdk
+   * database file is backed up within our temporary directory path. If some
+   * error occurs, `this.revertDatbase()` will be called to revert the backup.
+   */
+  private async backupDatabase() {
+    const src = this.dbPath
+    const dest = this.backupFile
+    this.agent.config.logger?.trace(`Creating backup from '${src}' to '${dest}'`)
+
+    // Create the directories for the backup
+    await this.fs.createDirectory(dest)
+
+    // Copy the supplied database to the backup destination
+    await this.fs.copyFile(src, dest)
+
+    if (!(await this.fs.exists(dest))) {
+      throw new IndySdkToAskarMigrationError('Could not locate the new backup file')
+    }
+  }
+
+  /**
+   * Reverts backed up database file to the original path, if its missing, and
+   * deletes the backup. We do some additional, possible redundant, exists checks
+   * here to be extra sure that only a happy flow occurs.
+   */
+  private async revertDatabase() {
+    // "Impossible" state. Since we do not continue if `this.backupDatabase()`
+    // fails, this file should always be there. If this error is thrown, we
+    // cannot correctly revert the state.
+    if (!(await this.fs.exists(this.backupFile))) {
+      throw new IndySdkToAskarMigrationError('Backup file could not be found while trying to revert the state')
+    }
+
+    /**
+     * Since we used `copy` to get the file, it should still be there. We
+     * double-check here to be sure.
+     */
+    if (!(await this.fs.exists(this.dbPath))) {
+      return
+    } else {
+      this.agent.config.logger?.trace(`Moving '${this.backupFile}' back to the original path: '${this.dbPath}`)
```

Comment:
```suggestion
      this.agent.config.logger.trace(`Moving '${this.backupFile}' back to the original path: '${this.dbPath}`)
```

Reply:
Seem this everywhere in the file, it's not needed right?

Reply:
I think it's leftover from some previous code of the logger. We should add an eslint rule to not use `?` if it can't be `null`/`undefined` :).